src/Digdir.Domain.Dialogporten.Infrastructure/Altinn/Authorization/AltinnAuthorizationClient.cs

```diff
@@ -140,9 +142,19 @@ private async Task<DialogDetailsAuthorizationResult> PerformDialogDetailsAuthori
     {
         var xacmlJsonRequest = DecisionRequestHelper.CreateDialogDetailsRequest(request);
         var xamlJsonResponse = await SendPdpRequest(xacmlJsonRequest, cancellationToken);
+        LogIfIndeterminate(xamlJsonResponse);
+
         return DecisionRequestHelper.CreateDialogDetailsResponse(request.AltinnActions, xamlJsonResponse);
     }
 
+    private void LogIfIndeterminate(XacmlJsonResponse? response)
+    {
+        if (response?.Response != null && response.Response.Any(result => result.Decision == "Indeterminate"))
+        {
+            _logger.LogError("Altinn Authorization returned an Indeterminate decision");
```

Comment:
What more do we get from this response? Is there any other data we can safely log for helping with debugging? 
(From either request or response)

Reply:
Added the xacml request to the log message. The response only contains the fact that it is Indeterminate.

Reply:
Example of xacml request in log:

`[09:24:50 ERR] Authorization request to authorization/api/v1/authorize returned decision Indeterminate. Request: {"Request":{"Resource":[{"Id":"r1","Attribute":[{"AttributeId":"urn:altinn:resource","Value":"ttd-dialogporten-automated-tests"},{"AttributeId":"urn:altinn:person:identifier-no","Value":"14886498226"},{"AttributeId":"urn:altinn:resourceinstance","Value":"b7979101-ec80-4470-9323-08ba1102e44a"},{"AttributeId":"urn:altinn:subresource","Value":"main"}]},{"Id":"r2","Attribute":[{"AttributeId":"urn:altinn:person:identifier-no","Value":"14886498226"},{"AttributeId":"urn:altinn:resource","Value":"banan"}]},{"Id":"r3","Attribute":[{"AttributeId":"urn:altinn:resource","Value":"ttd-dialogporten-automated-tests"},{"AttributeId":"urn:altinn:person:identifier-no","Value":"14886498226"},{"AttributeId":"urn:altinn:resourceinstance","Value":"b7979101-ec80-4470-9323-08ba1102e44a"},{"AttributeId":"urn:altinn:subresource","Value":"elementius"}]}],"Action":[{"Id":"a1","Attribute":[{"AttributeId":"urn:oasis:names:tc:xacml:1.0:action:action-id","Value":"read"}]},{"Id":"a2","Attribute":[{"AttributeId":"urn:oasis:names:tc:xacml:1.0:action:action-id","Value":"sit"}]},{"Id":"a3","Attribute":[{"AttributeId":"urn:oasis:names:tc:xacml:1.0:action:action-id","Value":"submit"}]},{"Id":"a4","Attribute":[{"AttributeId":"urn:oasis:names:tc:xacml:1.0:action:action-id","Value":"transmissionread"}]}],"AccessSubject":[{"Id":"s1","Attribute":[{"AttributeId":"urn:altinn:authenticatemethod","Value":"NotDefined"},{"AttributeId":"urn:altinn:authlevel","Value":"3"},{"AttributeId":"urn:altinn:person:identifier-no","Value":"14886498226"}]}],"MultiRequests":{"RequestReference":[{"ReferenceId":["s1","r1","a1"]},{"ReferenceId":["s1","r2","a1"]},{"ReferenceId":["s1","r1","a2"]},{"ReferenceId":["s1","r1","a3"]},{"ReferenceId":["s1","r1","a3"]},{"ReferenceId":["s1","r3","a4"]}]}}}
`

Reply:
Updated after removal of Pid:

{
    "Request": {
        "ReturnPolicyIdList": false,
        "CombinedDecision": false,
        "XPathVersion": null,
        "Category": null,
        "Resource": [
            {
                "CategoryId": null,
                "Id": "r1",
                "Content": null,
                "Attribute": [
                    {
                        "AttributeId": "urn:altinn:resource",
                        "Value": "ttd-dialogporten-automated-tests",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:person:identifier-no",
                        "Value": "",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:resourceinstance",
                        "Value": "b7979101-ec80-4470-9323-08ba1102e44a",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:subresource",
                        "Value": "main",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    }
                ],
                "$type": "XacmlJsonCategory"
            },
            {
                "CategoryId": null,
                "Id": "r2",
                "Content": null,
                "Attribute": [
                    {
                        "AttributeId": "urn:altinn:person:identifier-no",
                        "Value": "",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:resource",
                        "Value": "banan",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    }
                ],
                "$type": "XacmlJsonCategory"
            },
            {
                "CategoryId": null,
                "Id": "r3",
                "Content": null,
                "Attribute": [
                    {
                        "AttributeId": "urn:altinn:resource",
                        "Value": "ttd-dialogporten-automated-tests",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:person:identifier-no",
                        "Value": "",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:resourceinstance",
                        "Value": "b7979101-ec80-4470-9323-08ba1102e44a",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:subresource",
                        "Value": "elementius",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    }
                ],
                "$type": "XacmlJsonCategory"
            }
        ],
        "Action": [
            {
                "CategoryId": null,
                "Id": "a1",
                "Content": null,
                "Attribute": [
                    {
                        "AttributeId": "urn:oasis:names:tc:xacml:1.0:action:action-id",
                        "Value": "read",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    }
                ],
                "$type": "XacmlJsonCategory"
            },
            {
                "CategoryId": null,
                "Id": "a2",
                "Content": null,
                "Attribute": [
                    {
                        "AttributeId": "urn:oasis:names:tc:xacml:1.0:action:action-id",
                        "Value": "sit",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    }
                ],
                "$type": "XacmlJsonCategory"
            },
            {
                "CategoryId": null,
                "Id": "a3",
                "Content": null,
                "Attribute": [
                    {
                        "AttributeId": "urn:oasis:names:tc:xacml:1.0:action:action-id",
                        "Value": "submit",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    }
                ],
                "$type": "XacmlJsonCategory"
            },
            {
                "CategoryId": null,
                "Id": "a4",
                "Content": null,
                "Attribute": [
                    {
                        "AttributeId": "urn:oasis:names:tc:xacml:1.0:action:action-id",
                        "Value": "transmissionread",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    }
                ],
                "$type": "XacmlJsonCategory"
            }
        ],
        "AccessSubject": [
            {
                "CategoryId": null,
                "Id": "s1",
                "Content": null,
                "Attribute": [
                    {
                        "AttributeId": "urn:altinn:authenticatemethod",
                        "Value": "NotDefined",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:authlevel",
                        "Value": "3",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    },
                    {
                        "AttributeId": "urn:altinn:person:identifier-no",
                        "Value": "",
                        "Issuer": null,
                        "DataType": null,
                        "IncludeInResult": false,
                        "$type": "XacmlJsonAttribute"
                    }
                ],
                "$type": "XacmlJsonCategory"
            }
        ],
        "RecipientSubject": null,
        "IntermediarySubject": null,
        "RequestingMachine": null,
        "MultiRequests": {
            "RequestReference": [
                {
                    "ReferenceId": [
                        "s1",
                        "r1",
                        "a1"
                    ],
                    "$type": "XacmlJsonRequestReference"
                },
                {
                    "ReferenceId": [
                        "s1",
                        "r2",
                        "a1"
                    ],
                    "$type": "XacmlJsonRequestReference"
                },
                {
                    "ReferenceId": [
                        "s1",
                        "r1",
                        "a2"
                    ],
                    "$type": "XacmlJsonRequestReference"
                },
                {
                    "ReferenceId": [
                        "s1",
                        "r1",
                        "a3"
                    ],
                    "$type": "XacmlJsonRequestReference"
                },
                {
                    "ReferenceId": [
                        "s1",
                        "r1",
                        "a3"
                    ],
                    "$type": "XacmlJsonRequestReference"
                },
                {
                    "ReferenceId": [
                        "s1",
                        "r3",
                        "a4"
                    ],
                    "$type": "XacmlJsonRequestReference"
                }
            ],
            "$type": "XacmlJsonMultiRequests"
        },
        "XForwardedForHeader": null,
        "$type": "XacmlJsonRequest"
    },
    "$type": "XacmlJsonRequestRoot"
}